tests/test_hooks_intelligence.py

```
@@ -0,0 +1,422 @@
+
+# ── brain_maintain ──
+
+from gradata.hooks.brain_maintain import main as maintain_main
+
+
+def test_brain_maintain_runs_silently(tmp_path):
+    lessons = tmp_path / "lessons.md"
+    lessons.write_text("[2026-04-01] [RULE:0.92] PROCESS: Plan first\n")
+
+    with patch.dict(os.environ, {"GRADATA_BRAIN_DIR": str(tmp_path)}):
+        with patch("gradata._query.fts_index") as mock_fts:
```

Ruff (0.15.9)

[warning] Use a single `with` statement with multiple contexts instead of nested `with` statements

(SIM117)